includes/woopay/class-woopay-session.php

```diff
@@ -533,9 +533,13 @@ private static function get_woopay_verified_email_address() {
 	 * @return bool True if request is a Store API request, false otherwise.
 	 */
 	private static function is_store_api_request(): bool {
-		$url_parts    = wp_parse_url( esc_url_raw( $_SERVER['REQUEST_URI'] ?? '' ) ); // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.MissingUnslash
-		$request_path = rtrim( $url_parts['path'], '/' );
-		$rest_route   = str_replace( trailingslashit( rest_get_url_prefix() ), '', $request_path );
+		if ( isset( $_REQUEST['rest_route'] ) ) {
```

Comment:
This conditional structure favors the `rest_route` parameter over the URI, are there concerns with that approach? For example, could a `rest_route` parameter be present, but we actually wanted to use the URI instead?

Reply:
It seems like it should be either the URL or the `rest_route` query parameter.

https://github.com/WordPress/WordPress/blob/39ccc3798dc77c98330d3d0f6ac87c1d556e38e1/wp-includes/rest-api/class-wp-rest-request.php#L1036-L1044